crates/astria-conductor/src/executor.rs

```diff
@@ -156,12 +161,23 @@ impl<C: ExecutionClient> Executor<C> {
                         );
                         continue;
                     };
-                    if let Err(e) = self.execute_block(block_subset).await {
-                        error!(
-                            height = height,
-                            error = ?e,
-                            "failed to execute block"
-                        );
+                    match self.execute_block(block_subset.clone()).await {
+                        Ok(Some(executed_block)) => {
+                            // set soft but leave firm the same
+                            self.update_commitment_state(
+                                self.commitment_state.firm.clone().unwrap(),
+                                executed_block,
+                            )
+                            .await?;
```

Comment:
```suggestion
                            .await.wrap_err("failed to update commitment state")?;
```

Reply:
actually, i think returning an error here stops the whole event loop and thus the entire conductor - is that what we want, or should we just log here? the error below is just logged, so maybe we do that?

Reply:
updated! to avoid nesting, i refactored the match into some `if lets`. please lmk if this is better @noot